app/admin/admin.py

```diff
@@ -38,16 +41,27 @@ class ApplicationView(ModelView):
 
 class ProviderView(ModelView):
     can_export = True
+    list_template = ('admin/providers.html'
+                     )  # Extending the list view to allow for CSV import
 
-    @expose('/import/', methods=['POST'])
+    @expose('/import', methods=['GET', 'POST'])
     def import_file(self):
         form = ProviderImportForm()
         if form.validate_on_submit():
-            # TODO: Check the contents for its API
-            file_content = form.file.data
-            # TODO: Parse the content and store it in db
+            # form.file.data returns a Workzeug FileStorage object, .stream returns the tempfile w/CSV data in bytearray
+            file_content = form.file.data.stream.read().decode('utf-8')
+            with StringIO(file_content) as csv_file:
+                csv_file_reader = csv.DictReader(csv_file,
+                                                 fieldnames=CSV_SCHEMA)
+                for index, item in enumerate(csv_file_reader):
```

Comment:
Index should be in the CSV file, we can't generate it ourselves here. As it is the primary key.